server/app/Models/Doc.php

```diff
@@ -609,144 +524,16 @@ public static function allOwnedBy($userId)
 
     public static function getAllValidSponsors()
     {
-        $userMeta = UserMeta::where('meta_key', '=', UserMeta::TYPE_INDEPENDENT_SPONSOR)
-                            ->where('meta_value', '=', 1)
-                            ->get();
-
-        $groups = Group::where('status', '=', Group::STATUS_ACTIVE)
-                        ->get();
-
-        $results = new Collection();
-
-        $userIds = array();
-
-        foreach ($userMeta as $m) {
-            $userIds[] = $m->user_id;
-        }
-
-        if (!empty($userIds)) {
-            $users = User::whereIn('id', $userIds)->get();
-
-            foreach ($users as $user) {
-                $row = array(
-                        'display_name' => "{$user->fname} {$user->lname}",
-                        'sponsor_type' => 'individual',
-                        'id' => $user->id,
-                );
-
-                $results->add($row);
-            }
-        }
-
-        foreach ($groups as $group) {
-            $row = array(
-                    'display_name' => $group->display_name,
-                    'sponsor_type' => 'group',
+        return Group
+            ::where('status', '=', Group::STATUS_ACTIVE)
+            ->get()
+            ->map(function ($group) {
+                return [
```

Comment:
This is a little weird for PHP, where an array of objects would usually be expected, instead of an array of arrays. You could just do [`->select('id', 'display_name')`](https://laravel.com/docs/5.1/queries#selects) here instead which would be more expected.